### PR TITLE
Adds experimental --ion-version option

### PR DESCRIPTION
**Pull request by popematt**
### Issue #, if available:

#85 

### Description of changes:

Changes are mostly broken up by commits

* Removes the beta namespace
* Removes the passthrough `dump` command in favor of using clap's `alias` feature to achieve the same effect.
* Adds `--unstable`/`-X` to opt into experimental features.
* Adds an `is_porcelain()` function to the `IonCliCommand` trait. If implemented to return true, it gets a warning at the bottom of the command help that the output is not intended to be suitable for machine-readable use cases.
* Adds `--ion-version` option

----
_**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,18 +203,25 @@ A collection of tools for working with Ion data.
...
-  beta     The 'beta' command is a namespace for commands whose interfaces are not yet stable.
-  cat      Prints all Ion input files to the specified output in the requested format.
-  head     Prints the specified number of top-level values in the input stream.
-  inspect  Displays hex-encoded binary Ion alongside its equivalent text Ion.
-               Its output prioritizes human readability and is likely to change
-               between versions. Stable output for programmatic use cases is a
-               non-goal.
-  help     Print this message or the help of the given subcommand(s)
+  cat        Prints all Ion input files to the specified output in the requested format.
+  count      (UNSTABLE) Prints the number of top-level values found in the input stream.
```
[the remark this answers is not in the document]
Comment (popematt) [reply README.md:207]: I have no idea, but it was in the `beta` namespace, so I'm leaving it as unstable for now. We can fix that in a separate PR.

File: src/bin/ion/commands/mod.rs [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+    /// Whether the output format is machine-readable.
+    ///
+    /// Commands that are "plumbing" should default to putting one output (result, value, document)
+    /// on each line in a machine-readable format (file name, Ion value(s), integers, booleans)
+    /// without any prose or table formatting, etc.
+    ///
+    /// See https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain#_plumbing_porcelain
+    fn is_porcelain(&self) -> bool {
+        false
+    }
```
[the remark this answers is not in the document]
Comment (popematt) [reply src/bin/ion/commands/mod.rs:43]: I don't mind flipping them.

**Review by jobarr-amzn — APPROVED** [review]
I'm happy to approve it as-is, I'd like to change the default unstable/porcelain values but there's no need to couple that here.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (popematt) [reply README.md:207]: Actually, it is possible that #104 could eventually replace `count`, so maybe that's a good reason to leave it a "unstable" for now.

**Pull request merged**